From 19a302414b5137e07581e75ccc2e07370080f6c9 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Fri, 15 Mar 2019 08:49:00 +0800
Subject: [PATCH] MLK-21148-1 arm64: dts: imx8qm: correct isi irq type setting

commit  09ff6f9d1de7a0c3a9a9d40068ed8f552542f817 from
https://source.codeaurora.org/external/imx/linux-imx.git

GIC irqchip driver introduces the warning when IRQ type is
set to IRQ_TYPE_NONE while it is NOT belonging to GIC_IRQ_TYPE_PARTITION,
the ISI irqs' setting trigger below WARNING during kernel boot
up, set ISI irqs type to IRQ_TYPE_LEVEL_HIGH to avoid such
warning during kernel bootup, the GIC driver ONLY supports
IRQ_TYPE_LEVEL_HIGH and IRQ_TYPE_EDGE_RISING for SPIs actually.

drivers/irqchip/irq-gic-v3.c:
WARN_ON(*type == IRQ_TYPE_NONE &&
	fwspec->param[0] != GIC_IRQ_TYPE_PARTITION);

[    0.650117] WARNING: CPU: 3 PID: 1 at drivers/irqchip/irq-gic-v3.c:975
	 gic_irq_domain_translate+0xa8/0x110
[    0.659434] Modules linked in:
[    0.662478] CPU: 3 PID: 1 Comm: swapper/0 Not tainted 4.19.26-04846-g3ebadee #216
[    0.669944] Hardware name: Freescale i.MX8QM MEK (DT)
[    0.674984] pstate: 00000005 (nzcv daif -PAN -UAO)
[    0.679759] pc : gic_irq_domain_translate+0xa8/0x110
[    0.684717] lr : irq_create_fwspec_mapping+0x64/0x33c
[    0.689747] sp : ffff00000806b890
[    0.693049] x29: ffff00000806b890 x28: ffff8008f81264c0
[    0.698349] x27: 0000000000000000 x26: 0000000000000001
[    0.703647] x25: ffff8008f81264c0 x24: ffff0000094b8000
[    0.708946] x23: 0000000000000000 x22: ffff00000806b908
[    0.714246] x21: ffff8008fff77d00 x20: ffff0000094b8000
[    0.719545] x19: ffff8008f9824000 x18: ffff00000923a000
[    0.724844] x17: ffff000009239000 x16: ffff000009239948
[    0.730144] x15: 0000000000000001 x14: ffffffffffffffff
[    0.735443] x13: 000000000000030f x12: 0000000000000000
[    0.740742] x11: 0000000000000030 x10: 0101010101010101
[    0.746042] x9 : 0000000000000006 x8 : 7f7f7f7f7f7f7f7f
[    0.751341] x7 : 0000000000000000 x6 : 0000000000000092
[    0.756641] x5 : ffff000008485f14 x4 : ffff000008f9a0a0
[    0.761940] x3 : ffff00000806b8cc x2 : 0000000000000000
[    0.767240] x1 : 00000000a110c8ee x0 : 0000000000000000
[    0.772542] Call trace:
[    0.774976]  gic_irq_domain_translate+0xa8/0x110
[    0.779580]  irq_create_of_mapping+0x80/0xa8
[    0.783841]  of_irq_get+0x70/0xe4
[    0.787138]  of_irq_to_resource+0x38/0xfc
[    0.791134]  of_irq_to_resource_table+0x4c/0x68
[    0.795655]  of_device_alloc+0x120/0x1b8
[    0.799562]  of_platform_device_create_pdata+0x7c/0x11c
[    0.804774]  of_platform_bus_create+0x140/0x388
[    0.809289]  of_platform_bus_create+0x1a0/0x388
[    0.813806]  of_platform_populate+0x50/0xd4
[    0.817983]  of_platform_default_populate_init+0xbc/0xd0
[    0.823280]  do_one_initcall+0x58/0x168
[    0.827104]  kernel_init_freeable+0x1b4/0x274
[    0.831446]  kernel_init+0x10/0x104
[    0.834921]  ret_from_fork+0x10/0x18
[    0.838482] ---[ end trace 52dbd1b75308ae5a ]---

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Xiaolei Wang <Xiaolei.Wang@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-device.dtsi    | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
index 3cea30587dc7..d7f9eb74971c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
@@ -2455,7 +2455,7 @@
 		isi_0: isi@58100000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58100000 0x0 0x10000>;
-			interrupts = <0 297 0>;
+			interrupts = <0 297 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <2 0 2>;  /* <Input MIPI_VCx Output>
 									Input:  0-DC0, 1-DC1, 2-MIPI CSI0, 3-MIPI CSI1, 4-HDMI, 5-MEM
 									VCx:    0-VC0, 1-VC1, 2-VC2, 3-VC3, MIPI CSI only
@@ -2471,7 +2471,7 @@
 		isi_1: isi@58110000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58110000 0x0 0x10000>;
-			interrupts = <0 298 0>;
+			interrupts = <0 298 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <2 1 2>;
 			clocks = <&clk IMX8QM_IMG_PDMA_1_CLK>;
 			clock-names = "per";
@@ -2484,7 +2484,7 @@
 		isi_2: isi@58120000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58120000 0x0 0x10000>;
-			interrupts = <0 299 0>;
+			interrupts = <0 299 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <2 2 2>;
 			clocks = <&clk IMX8QM_IMG_PDMA_2_CLK>;
 			clock-names = "per";
@@ -2497,7 +2497,7 @@
 		isi_3: isi@58130000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58130000 0x0 0x10000>;
-			interrupts = <0 300 0>;
+			interrupts = <0 300 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <2 3 2>;
 			clocks = <&clk IMX8QM_IMG_PDMA_3_CLK>;
 			clock-names = "per";
@@ -2510,7 +2510,7 @@
 		isi_4: isi@58140000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58140000 0x0 0x10000>;
-			interrupts = <0 301 0>;
+			interrupts = <0 301 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <3 0 2>;
 			clocks = <&clk IMX8QM_IMG_PDMA_4_CLK>;
 			clock-names = "per";
@@ -2523,7 +2523,7 @@
 		isi_5: isi@58150000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58150000 0x0 0x10000>;
-			interrupts = <0 302 0>;
+			interrupts = <0 302 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <3 1 2>;
 			clocks = <&clk IMX8QM_IMG_PDMA_5_CLK>;
 			clock-names = "per";
@@ -2536,7 +2536,7 @@
 		isi_6: isi@58160000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58160000 0x0 0x10000>;
-			interrupts = <0 303 0>;
+			interrupts = <0 303 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <3 2 2>;
 			clocks = <&clk IMX8QM_IMG_PDMA_6_CLK>;
 			clock-names = "per";
@@ -2549,7 +2549,7 @@
 		isi_7: isi@58170000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58170000 0x0 0x10000>;
-			interrupts = <0 304 0>;
+			interrupts = <0 304 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <3 3 2>;
 			clocks = <&clk IMX8QM_IMG_PDMA_7_CLK>;
 			clock-names = "per";
-- 
2.17.1

